### PR TITLE
[EPMRPP-42845] SAML, LDAP, AD single interface

### PR DESCRIPTION
**Pull request by ihar-kahadouski**

### CODE REVIEW

**Comment by pbortnik** [comment]
misha vsio huinia dava po novoy

File: src/main/java/com/epam/reportportal/auth/integration/handler/CreateOrUpdateIntegrationStrategy.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
+package com.epam.reportportal.auth.integration.handler;
```
Comment (pbortnik) [inline src/main/java/com/epam/reportportal/auth/integration/handler/CreateOrUpdateIntegrationStrategy.java]: headers

File: src/main/java/com/epam/reportportal/auth/integration/handler/impl/CreateOrUpdateActiveDirectoryIntegrationStrategy.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
+package com.epam.reportportal.auth.integration.handler.impl;
```
Comment (pbortnik) [inline src/main/java/com/epam/reportportal/auth/integration/handler/impl/CreateOrUpdateActiveDirectoryIntegrationStrategy.java]: headers

File: src/main/java/com/epam/reportportal/auth/integration/builder/SamlBuilder.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package com.epam.reportportal.auth.integration.builder;
```
[the remark this answers is not in the document]
Comment (ihar-kahadouski) [reply src/main/java/com/epam/reportportal/auth/integration/builder/SamlBuilder.java]: done

File: src/main/java/com/epam/reportportal/auth/integration/handler/CreateOrUpdateIntegrationStrategy.java (hunk 1 above)
Comment (ihar-kahadouski) [reply src/main/java/com/epam/reportportal/auth/integration/handler/CreateOrUpdateIntegrationStrategy.java]: done

File: src/main/java/com/epam/reportportal/auth/integration/handler/impl/CreateOrUpdateActiveDirectoryIntegrationStrategy.java (hunk 2 above)
Comment (ihar-kahadouski) [reply src/main/java/com/epam/reportportal/auth/integration/handler/impl/CreateOrUpdateActiveDirectoryIntegrationStrategy.java]: done

**Pull request merged**